d3a_interface/sim_results/cumulative_grid_trades.py

```diff
@@ -184,14 +184,18 @@ def _update_area_children_in_accumulated_trades_dict(cls, accumulated_trades, ar
             area: area dict that includes children
         Returns: None
         """
-        current_child_names = [
-            c['name'] for c in accumulated_trades[area['uuid']]['children']
+        current_child_uuids = [
+            c['uuid'] for c in accumulated_trades[area['uuid']]['children']
         ]
         for child in area.get("children", []):
-            if child["name"] not in current_child_names:
+            if child["uuid"] not in current_child_uuids:
                 accumulated_trades[area['uuid']]['children'].append(
                     cls._generate_accumulated_trades_child_dict(accumulated_trades, child)
                 )
+            else:
+                child_index = current_child_uuids.index(child['uuid'])
```

Comment:
IMO you should use ```enumerate``` , this way you have the index of the current iteration without invoking list.index() , which takes O(N) in each time called

Reply:
Update, I see that you're not looping over the same list, so my proposal doesn't work